File: year_1/PROGKI/Problem set 2/caesar.c
/*
 * caesar.c
 *
 * Introduction Programming Artificial Intelligence
 * Tim Stolp
 *
 * asks for encryption number.
 * Asks for message.
 * Prints encrypted message.
 *
 * changes every character in the message by the number of places given by the encryption number.
 *
 */

#include <stdio.h>
#include <string.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    // if argument 1 is null or number of arguments bigger than 2
    if(argv[1] == NULL || argc > 2)
    {
        // print error message
        printf("Usage: ./caesar *number*\n");
        // return error number 1
        return 1;
    }

    // turn argument 1 into integer
    int key = atoi(argv[1]);
    // get string from user
    string s = get_string("plaintext: ");
    // get stringlength
    int l = strlen(s);

    // print encrypted text
    printf("ciphertext: ");
    for(int i = 0; i < l; i++)
    {
        // if character is space, print space
        if(s[i] == ' ')
        {
            printf(" ");
        }
        else
        {
            int k, letter;
            // do "key % 26" amount of times
            for(k = 0, letter = s[i]; k < key % 26; k++)
            {
                // if character is Z or z do character - 25
                if(letter == 90 || letter == 122)
                {
                    letter = letter - 25;
                }
                // do character + 1
                else
                {
                    letter = letter + 1;
                }
            }
            // print integer as character
            printf("%c", letter);
        }
    }
    // next line
    printf("\n");
}

File: year_1/PROGKI/Problem_Set_4/caesar.py
#
# caesar.py
#
# Artificial Intelligence
# Tim Stolp
#
# encrypt a message using caesar's encryption
#
# encrypt text using a number as encryption key
#
#####################################

import sys

# if amount of arguments is not 2 print usage message and exit
if len(sys.argv) != 2:
    print("usage: python caesar.py key")
    exit(1)

# ask text to encrypt
plaintext = input("plaintext: ")

# print this
print("cyphertext: ", end="")

# go through plaintext
# with each letter move it a certain amount of places through the alphabet
# print encrypted text
for i in plaintext:
    letter = ord(i)
    for j in range(int(sys.argv[1])):
        if letter == ord("z") or letter == ord("Z"):
            letter = letter - 25
        else:
            letter = letter + 1
    print(chr(letter), end="")

# next line
print()
